core/vnl/vnl_vector_fixed_ref.h

```diff
@@ -90,13 +90,13 @@ class VNL_EXPORT vnl_vector_fixed_ref_const
   // for vnl_vector_ref but not for vnl_vector_fixed_ref. There is also a
   // conversion operator that should work most of the time.
   // \sa vnl_vector_ref::non_const
-  const vnl_vector_ref<T> as_ref() const { return vnl_vector_ref<T>( n, const_cast<T*>(data_) ); }
+  vnl_vector_ref<T> as_ref() const { return vnl_vector_ref<T>( n, const_cast<T*>(data_) ); }
 
   //: Cheap conversion to vnl_vector_ref
   // Sometimes, such as with templated functions, the compiler cannot
   // use this user-defined conversion. For those cases, use the
   // explicit as_ref() method instead.
-  operator const vnl_vector_ref<T>() const { return vnl_vector_ref<T>( n, const_cast<T*>(data_) ); }
```

Comment:
As I commented before: the unusual 'const'-qualification at the top level of the return type appears to be meant to prevent modifications of the original object (vnl_vector_fixed, in this case) through the "ref" object.

Reply:
Reverted:  FYI That is part of another larger issue that I want to keep separate.

the larger issue is my concern about excessive unnecessary implicit type conversions.